src/server/macos.rs

```diff
@@ -773,6 +783,17 @@ pub fn create_launchctl_service(name: &str, meta: &Metadata)
     }
 }
 
+pub fn recreate_launchctl_service(name: &str, meta: &Metadata)
+    -> anyhow::Result<()>
+{
+    bootout_launchctl_service(name)?;
+    if meta.start_conf == StartConf::Auto {
```

Comment:
I think this should be conditional on `is_service_loaded` rather than `start_conf`.  If the service was loaded it should stay loaded after the upgrade.

Reply:
Yes I agree! It's just a little bit more complex than that, because a loaded service may not be running. And I think we are expecting the running state to be reserved after the upgrade. So we have 6 situations:

| Auto?  | Loaded? | Running? | Expected Result                                           |
|--------|---------|----------|-----------------------------------------------------------|
| auto   | Yes     | Yes      | loaded + running                                          |
| auto   | Yes     | No       | loaded + stopped                                          |
| auto   | No      | No       | loaded + stopped (assuming auto service is always loaded) |
| manual | Yes     | Yes      | loaded + running                                          |
| manual | Yes     | No       | not running                                               |
| manual | No      | No       | not running                                               |

Means we shall also check if the instance was running before the upgrade. I'll push a fix for this.

Reply:
Loading the new service will automatically start the server, and unfortunately we don't have a reliable-yet-easy way to wait to the right moment to send the SIGTERM to achieve the "loaded + stopped" result. I'm currently just sleeping 10s because it is about the time for startup, and this is a relatively rare case (auto-starting service manually stopped before migration).

Reply:
The table is updated - no more "loaded + stopped" results now.